### PR TITLE
Refactor install code and fix test

### PR DESCRIPTION
**Pull request by larowlan**
Addressing items from [my own review](https://www.drupal.org/project/drupal/issues/2809635#comment-12403176)

Fixes items 2-6 and 8 from review.

Also makes test pass, (when umami theme is present)

### CODE REVIEW

File: demo_umami_content/src/InstallHelper.php [hunk 1]
```diff
@@ -0,0 +1,368 @@
+<?php
+
+namespace Drupal\demo_umami_content;
+
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\State\StateInterface;
+use Drupal\user\Entity\User;
```
Comment (navneet0693) [inline demo_umami_content/src/InstallHelper.php]: This can be removed. This is an unused class.

**Comment by gareth-fivemile** [comment]
Just taking a look through this now.

**Comment by gareth-fivemile** [comment]
Great work, thanks @larowlan!

I removed the dependency on migrate too, thanks @JayKandari.
Also took care of the unused User class, thanks @navneet0693.

File: demo_umami_content/demo_umami_content.install [hunk 2]
```diff
@@ -2,255 +2,23 @@
...
  * Implements hook_install().
  */
 function demo_umami_content_install() {
-  // Install default articles.
-  _install_demo_umami_content_articles();
-  // Install default recipes.
-  _install_demo_umami_content_recpies();
-  // Install default pages.
-  _install_demo_umami_content_pages();
+  \Drupal::classResolver()->getInstanceFromDefinition(InstallHelper::class)->importArticles()
```
Comment (jibran) [inline demo_umami_content/demo_umami_content.install]: This should be wrapped in config syncing check.

Comment (gareth-fivemile) [reply demo_umami_content/demo_umami_content.install]: This has been done:
https://github.com/gareth-fivemile/demo_umami/pull/29

**Pull request merged**